src/main/cpp/src/murmur_hash.cu

```diff
@@ -142,9 +122,32 @@ class murmur_device_row_hasher {
     }
 
     Nullate const _check_nulls;  ///< Whether to check for nulls
-    uint32_t const _seed;        ///< The seed to use for hashing, also returned for null elements
+    result_type const _seed;     ///< The seed to use for hashing, also returned for null elements
   };
 
+  using result_type = typename element_hasher_adapter<hash_function>::result_type;
+
+ public:
+  /**
+   * @brief Return the hash value of a row in the given table.
+   *
+   * @param row_index The row index to compute the hash value of
+   * @return The hash value of the row
+   */
+  __device__ auto operator()(cudf::size_type row_index) const noexcept
+  {
+    return cudf::detail::accumulate(
+      _table.begin(),
+      _table.end(),
+      _seed,
+      cuda::proclaim_return_type<murmur_hash_value_type>(
+        [row_index, nulls = this->_check_nulls] __device__(auto hash, auto column) {
+          return cudf::type_dispatcher(
+            column.type(), element_hasher_adapter<hash_function>{nulls, hash}, column, row_index);
+        }));
+  }
```

Comment:
I'm checking if I can minimize the change further.  Maybe there isn't a need to move the definition of `element_hasher_adapter`.

Reply:
Ok, minimized the change, I think.